hack/lib/golang.sh

```diff
@@ -49,6 +49,7 @@ readonly KUBE_SUPPORTED_CLIENT_PLATFORMS=(
   linux/s390x
   linux/ppc64le
   darwin/amd64
+  darwin/arm
```

Comment:
this must be arm64 but not arm.

also this is not the only place you should add, there are `darwin/amd64` in other lines of this file, and in `test/typecheck/main.go` as well.

Reply:
Ref. https://gist.github.com/inductor/d944f90cb277d077fb8b737ff74b9cf2

Reply:
@inductor just find one M1 to test it.
This is still work in process. Thanks for your comments.

Reply:
Yeah I know :) I made it work with the patch.